src/components/MAPI/workernodes/__tests__/WorkerNodesNodePoolItem.tsx

```diff
@@ -0,0 +1,431 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import sub from 'date-fns/fp/sub';
+import { createMemoryHistory } from 'history';
+import TestOAuth2 from 'lib/OAuth2/TestOAuth2';
+import * as metav1 from 'model/services/mapi/metav1';
+import nock from 'nock';
+import React from 'react';
+import { Providers, StatusCodes } from 'shared/constants';
+import { cache, SWRConfig } from 'swr';
+import * as capiexpv1alpha3Mocks from 'testUtils/mockHttpCalls/capiv1alpha3/exp';
+import * as capzexpv1alpha3Mocks from 'testUtils/mockHttpCalls/capzv1alpha3/exp';
+import { getComponentWithStore } from 'testUtils/renderUtils';
+
+import { getAdditionalColumns } from '../ClusterDetailWorkerNodes';
+import WorkerNodesNodePoolItem from '../WorkerNodesNodePoolItem';
+
+function getComponent(
+  props: React.ComponentPropsWithoutRef<typeof WorkerNodesNodePoolItem>
+) {
+  const history = createMemoryHistory();
+  const auth = new TestOAuth2(history, true);
+
+  const Component = (p: typeof props) => (
+    <SWRConfig value={{ dedupingInterval: 0 }}>
+      <WorkerNodesNodePoolItem {...p} />
+    </SWRConfig>
+  );
+
+  return getComponentWithStore(
+    Component,
+    props,
+    undefined,
+    undefined,
+    history,
+    auth
+  );
+}
+
+describe('WorkerNodesNodePoolItem', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  afterEach(() => {
+    cache.clear();
+  });
+
+  it('renders without crashing', () => {
+    render(getComponent({}));
+  });
+
+  it('displays a loading animation if the node pool is not loaded yet', () => {
+    render(getComponent({}));
+
+    expect(screen.getAllByLabelText('Loading...').length).toEqual(8);
+  });
+
+  it('displays if a node pool has been deleted', () => {
+    const deletionDate = sub({
+      hours: 1,
+    })(new Date());
+
+    render(
+      getComponent({
+        nodePool: {
+          ...capiexpv1alpha3Mocks.randomCluster1MachinePool1,
+          metadata: {
+            ...capiexpv1alpha3Mocks.randomCluster1MachinePool1.metadata,
+            deletionTimestamp: deletionDate.toISOString(),
+          },
+        },
+        providerNodePool: capzexpv1alpha3Mocks.randomCluster1AzureMachinePool1,
+      })
+    );
+
+    expect(
+      screen.getByText((_, node) => {
+        if (!node) return false;
+
+        const hasText = (n: Element) =>
+          n.textContent === 'Deleted about 1 hour ago';
+        const hasTextInChildren = Array.from(node.children).some((child) =>
+          hasText(child)
+        );
+
+        return hasText(node) && !hasTextInChildren;
+      })
+    ).toBeInTheDocument();
```

Comment:
This needs an assertion utility from #2551

I will replace this code with that once the other PR gets merged